spec/support/model_helpers.rb

```diff
@@ -0,0 +1,35 @@
+# use like this:
+# it { is_expected.to have_validation :valid_date_presence,
+#                                     :on => :start_date,
+#                                     :options => [:allow_blank, :allow_nil]
+#                                     }
+RSpec::Matchers.define :have_validation do |validator, params|
+  validator_class = eval(validator.to_s.camelize + "Validator")
+
+  match do |model|
+    @_validators = model.class.validators.select { |v| (v.class == validator_class) && (v.attributes.include? params[:on]) }
+    validator_included_once? && (validator_receives_correct_options? params[:options])
```

Comment:
This looks good, great job!
Small question: Can `validator_receives_correct_options?` not access params directly, in which case passing `params[:options]` as an argument is unnecessary?